contrib/epee/include/net/levin_protocol_handler_async.h

```diff
@@ -764,7 +765,44 @@ void async_protocol_handler_config<t_connection_context>::del_out_connections(si
 		}
 		--count;
 	}
-	
+
+	CRITICAL_REGION_END();
+}
+//------------------------------------------------------------------------------------------
+template<class t_connection_context>
+void async_protocol_handler_config<t_connection_context>::del_in_connections(size_t count)
+{
+	std::vector <boost::uuids::uuid> in_connections;
+	CRITICAL_REGION_BEGIN(m_connects_lock);
+	for (auto& c: m_connects)
+	{
+		if (c.second->m_connection_context.m_is_income)
+			in_connections.push_back(c.first);
+	}
+
+	if (in_connections.size() == 0)
+		return;
+
+	// close random in connections
+	unsigned seed = std::chrono::system_clock::now().time_since_epoch().count();
+	shuffle(in_connections.begin(), in_connections.end(), std::default_random_engine(seed));
+	while (count > 0 && in_connections.size() > 0)
+	{
+		try
+		{
+			auto i = in_connections.begin();
+			async_protocol_handler<t_connection_context> *conn = m_connects.at(*i);
+			del_connection(conn);
+			close(*i);
+			in_connections.erase(i);
+		}
+		catch (const std::out_of_range &e)
+		{
+			MWARNING("Connection not found in m_connects, continuing");
+		}
+		--count;
+	}
+
```

Comment:
Please factor with the existing function, should be pretty much all the same execpt the selection condition near the top.

Reply:
I factored out the deletion of the connections, but my C++ is too rusty to factor it further.